x-pack/plugins/lens/public/shared_components/legend_settings_popover.tsx

```diff
@@ -97,6 +120,44 @@ export interface LegendSettingsPopoverProps {
   groupPosition?: ToolbarButtonProps['groupPosition'];
 }
 
+const DEFAULT_TRUNCATE_LINES = 1;
+const MAX_TRUNCATE_LINES = 5;
+const MIN_TRUNCATE_LINES = 1;
+
+export const MaxLinesInput = ({
+  value,
+  setValue,
+  isDisabled,
+}: {
+  value: number;
+  setValue: (value: number) => void;
+  isDisabled: boolean;
+}) => {
+  const { inputValue, handleInputChange } = useDebouncedValue({ value, onChange: setValue });
+  return (
+    <EuiFieldNumber
+      data-test-subj="lens-legend-max-lines-input"
+      value={inputValue}
+      min={MIN_TRUNCATE_LINES}
+      max={MAX_TRUNCATE_LINES}
+      compressed
+      disabled={isDisabled}
+      onChange={(e) => {
+        const val = Number(e.target.value);
+        // we want to automatically change the values to the limits
+        // if the user enters a value that is outside the limits
+        if (val > MAX_TRUNCATE_LINES) {
```

Comment:
nit: Marco once shortened my code with this, less readable but you've left comment anyway :) Still up to you! : 
`onChange(Math.min(MAX_TRUNCATE_LINES, Math.max(val, MIN_TRUNCATE_LINES)));`